beacon-chain/blockchain/process_block.go

```diff
@@ -452,14 +449,6 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []interfaces.SignedBeac
 	if err := s.cfg.ForkChoiceStore.InsertNode(ctx, preState, lastBR); err != nil {
 		return errors.Wrap(err, "could not insert last block in batch to forkchoice")
 	}
-	// Prune forkchoice store only if the new finalized checkpoint is higher
```

Comment:
This is one of the examples that is much nicer now: we spent an entire morning finding this bug with @nisdas on Ropsten, this could have never happened if we had this logic within forkchoice.